hoot-test/src/main/cpp/hoot/test/main.cpp

```diff
@@ -155,43 +155,41 @@ class HootTestListener : public CppUnit::TestListener
   double _slowTest;
 };
 
-void filterPattern(CppUnit::Test* from, std::vector<CppUnit::Test*> &to, QString pattern,
-  bool includeOnMatch)
+void getTestVector(CppUnit::Test* from, vector<CppUnit::Test*>& to)
 {
-  QRegExp regex(pattern);
-
   for (int i = 0; i < from->getChildTestCount(); i++)
   {
     CppUnit::Test* child = from->getChildTestAt(i);
-    QString name = QString::fromStdString(child->getName());
     if (dynamic_cast<CppUnit::TestComposite*>(child))
-    {
-      filterPattern(child, to, pattern, includeOnMatch);
-    }
-    else if (regex.exactMatch(name) == includeOnMatch)
-    {
+      getTestVector(child, to);
+    else
+      to.push_back(child);
+  }
+}
+
+void getTestVector(const vector<TestPtr>& from, vector<CppUnit::Test*>& to)
+{
+  for (vector<TestPtr>::size_type i = 0; i < from.size(); ++i)
+  {
+    CppUnit::Test* child = from[i].get();
+    if (dynamic_cast<CppUnit::TestComposite*>(child))
+      getTestVector(child, to);
+    else
       to.push_back(child);
-    }
   }
 }
 
-void filterPattern(const std::vector<TestPtr> &from, std::vector<CppUnit::Test*> &to, QString pattern,
-  bool includeOnMatch)
+void filterPattern(const std::vector<CppUnit::Test*> &from, std::vector<CppUnit::Test*> &to,
+                   QString pattern, bool includeOnMatch)
 {
   QRegExp regex(pattern);
 
   for (size_t i = 0; i < from.size(); i++)
   {
-    CppUnit::Test* child = from[i].get();
+    CppUnit::Test* child = from[i];
     QString name = QString::fromStdString(child->getName());
-    if (dynamic_cast<CppUnit::TestComposite*>(child))
-    {
-      filterPattern(child, to, pattern, includeOnMatch);
-    }
-    else if (regex.exactMatch(name) == includeOnMatch)
-    {
+    if (regex.exactMatch(name) == includeOnMatch)
       to.push_back(child);
-    }
   }
 }
```

Comment:
Filtering doesn't need to be recursive and only hinges on the regex match equaling `includeOnMatch`